OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -74,6 +77,9 @@ void INotifyCreated.Created(Actor self)
 
 		void INotifyKilled.Killed(Actor self, AttackInfo e)
 		{
+			if (IsTraitDisabled)
+				return;
```

Comment:
Neither `IsTraitDisabled`, nor `!self.IsInWorld` is a complex Boolean expression; perhaps it would be fine to just use `IsTraitDisabled || !self.IsInWorld `.